packages/zero-solid/src/use-query.ts

```diff
@@ -1,37 +1,35 @@
-import {type Accessor, createMemo, onCleanup} from 'solid-js';
+import {type Accessor, onCleanup} from 'solid-js';
 import type {
-  Query,
   AdvancedQuery,
+  Query,
   QueryType,
   Smash,
   TableSchema,
 } from '../../zero-advanced/src/mod.js';
-import {solidViewFactory} from './solid-view.js';
 import type {ResultType} from '../../zql/src/query/typed-view.js';
+import {solidViewFactory} from './solid-view.js';
 
 export type QueryResultDetails = Readonly<{
   type: ResultType;
 }>;
 
 export type QueryResult<TReturn extends QueryType> = readonly [
-  Smash<TReturn>,
-  QueryResultDetails,
+  Accessor<Smash<TReturn>>,
+  Accessor<QueryResultDetails>,
 ];
 
 export function useQuery<
   TSchema extends TableSchema,
   TReturn extends QueryType,
->(querySignal: () => Query<TSchema, TReturn>): Accessor<QueryResult<TReturn>> {
-  return createMemo(() => {
-    const query = querySignal();
-    const view = (query as AdvancedQuery<TSchema, TReturn>).materialize(
-      solidViewFactory,
-    );
-
-    onCleanup(() => {
-      view.destroy();
-    });
+>(querySignal: () => Query<TSchema, TReturn>): QueryResult<TReturn> {
+  const query = querySignal();
```

Comment:
I'm worried without the use of createMemo that this function body can be run multiple times without querySignal's value changing. 

https://docs.solidjs.com/concepts/derived-values/memos

Reply:
or maybe even better
```
export function useQuery<
  TSchema extends TableSchema,
  TReturn extends QueryType,
>(
  querySignal: () => Query<TSchema, TReturn>,
): [Accessor<Smash<TReturn>>, Accessor<QueryResultDetails>] {
  const memoed = createMemo(() => {
    const query = querySignal();
    const view = (query as AdvancedQuery<TSchema, TReturn>).materialize(
      solidViewFactory,
    );

    onCleanup(() => {
      view.destroy();
    });

    return [view.data, {type: view.resultType}] as const;
  });

  return [() => memoed()[0], () => memoed()[1]];
}
```

Reply:
Looking at this I think I see a bug with the previous code which may be related to the iloop.  `view.data` was accessed in the `createMemo` and `view.data` is a property of a solid Store, thus a reactive value.  

So I think anytime the value of `view.data` changed (as determined using ===), the `createMemo` body was rerun, materializing a new view.